nilearn/_utils/niimg.py

```diff
@@ -79,21 +99,9 @@ def new_img_like(ref_img, data, affine, copy_header=False):
     new_img: image
         An image which has the same type as the reference image.
     """
-    if data.dtype == bool:
-        default_dtype = np.int8
-        if (LooseVersion(nibabel.__version__) >= LooseVersion('1.2.0') and
-                isinstance(ref_img, nibabel.freesurfer.mghformat.MGHImage)):
-            default_dtype = np.uint8
-        data = as_ndarray(data, dtype=default_dtype)
-    header = None
-    if copy_header:
-        header = copy.copy(ref_img.get_header())
-        header['scl_slope'] = 0.
-        header['scl_inter'] = 0.
-        header['glmax'] = 0.
-        header['cal_max'] = np.max(data) if data.size > 0 else 0.
-        header['cal_max'] = np.min(data) if data.size > 0 else 0.
-    return ref_img.__class__(data, affine, header=header)
+    from .niimg_conversions import check_niimg  # avoid circular import
```

Comment:
Can you be more explicit: what refactoring are you suggesting exactly (I am asking to make it more clear what need to be done on the PR before merging).